synfig-core/src/synfig/timepointcollect.cpp

```diff
@@ -88,7 +89,7 @@ synfig::waypoint_collect(std::set<Waypoint, std::less<UniqueID> >	&waypoint_set,
 		Layer::DynamicParamList::const_iterator iter;
 		int ret(0);
 		etl::handle<Layer_PasteCanvas> p = etl::handle<Layer_PasteCanvas>::cast_dynamic(layer);
-		if (!p){
+		if ((!p && initial_waypoint) || !initial_waypoint){
```

Comment:
hmm, yes but there was a cast here anyways as it is used in line 100, so I just moved the cast higher here so it can be used here and there. is there something i'm missing ?

i mean if this is not the initial waypoint yes the cast wont be needed on line 92 but it will still be needed for line 100. But if it is the intial waypoint then we will benefit from not having to cast twice.